pkg/resourcemanager/cosmosdb/sqldatabase/resource_differ.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package sqldatabase
+
+import "github.com/Azure/azure-sdk-for-go/services/cosmos-db/mgmt/2021-03-15/documentdb"
+
+func doResourcesMatch(expected *documentdb.SQLDatabaseResource, actual *documentdb.SQLDatabaseGetPropertiesResource) bool {
+	var expectedIDCoalesced *string
+	if expected != nil {
+		expectedIDCoalesced = expected.ID
+	}
+
+	var actualIDCoalesced *string
+	if actual != nil {
+		actualIDCoalesced = actual.ID
+	}
+
+	if (expectedIDCoalesced == nil) != (actualIDCoalesced == nil) {
+		return false
+	}
+	if expectedIDCoalesced != nil && actualIDCoalesced != nil && *expectedIDCoalesced != *actualIDCoalesced {
+		return false
+	}
+
+	return true
```

Comment:
```suggestion
	if expectedIDCoalesced == nil && actualIDCoalesced == nil {
		return true
	}

	return *expectedIDCoalesced == *actualIDCoalesced
```

Reply:
So the reason I didn't have any `return true`'s in this code is because it's really easy to accidentally write "these two match, we're good!" and forget that you're examining an object with 4 properties and you've only checked 2 of them. So I was striving the for "find mismatches, else if none true" to avoid that as I found myself making that mistake and then catching it like 2-3 times while writing these.

If folks don't think that's valuable in these cases (where there's only a single field) then I can change to do as you say. It felt like if another field was added to the struct later and somebody came into this method though, it'd be really easy to just "add stuff to the bottom" and not realize that actually it's wrong you need to go get rid of that `return true` as well.

Reply:
Unless any objection, will leave it as is for future extensibility and hopefully reduction in bugs.

Reply:
Yeah, that reasoning makes sense to me.